src/main/java/frc/robot/poseestimator/PoseEstimatorMath.java

```diff
@@ -0,0 +1,38 @@
+package frc.robot.poseestimator;
+
+import edu.wpi.first.math.geometry.*;
+
+public class PoseEstimatorMath {
+
+	public static double getKalmanRatio(double odometryStandardDeviation, double visionStandardDeviation) {
+		return odometryStandardDeviation / (odometryStandardDeviation + visionStandardDeviation);
+	}
+
+	public static Rotation2d combineVisionHeadingToGyro(
+		Rotation2d visionEstimatedHeading,
+		Rotation2d sampleDifferenceFromHeading,
+		Rotation2d currentEstimatedHeading,
+		double gyroStandardDeviation,
+		double visionStandardDeviation
+	) {
+		double combinedStandardDeviation = getKalmanRatio(gyroStandardDeviation, visionStandardDeviation);
+		Rotation2d estimatedHeadingAtSampleTime = currentEstimatedHeading.plus(sampleDifferenceFromHeading);
```

Comment:
So you've done this calculation: 
```estimatedHeadin - |estimatedHeading - gyroSample|```

Which is just the gyrosmaple. 
I think you want to add the difference between the samples and not just using the gyro estimation at timestamp.